platform/src/ToolsManager.js

```diff
@@ -178,6 +186,37 @@ class ToolManager {
         return errors;
     }
 
+    /**
+     * Reads toocl config text and replaces {{BASE-URL}} placeholders with the tool URL
+     * @returns string
+     */
+    rewriteUrl(base_url, toolUrl, tool_config_string) {
+        let tool_config_regexp = tool_config_string.match(new RegExp(/{{BASE-URL}}(:[0-9]+)?/g));
+        var toolConfig = tool_config_string;
+
+        if (tool_config_regexp != null){
+            // There are references to another endpoint, so base URL must be set accordingly (There is a {{BASE-URL}} placeholder plus a port).
```

Comment:
```suggestion
            // There are references to another endpoint, so base URL must be set accordingly (There is a {{BASE-URL}} placeholder plus an optional port).
```

Reply:
Actually, looking further down, it looks like you really want to make sure your regexp only matches `{{BASE-URL}}:<port>` because that then leaves only the "un-ported" `{{BASE-URL}}` cases.

Reply:
The reason to match the {{BASE-URL}}:port first is that any occurrences of these will be replaced with actual URLs before we get to occurrences of {{BASE-URL}} only. If done the other way round, `string.replaceAll()` function in JS will replace all instances of a placeholder and we cannot use the ports to determine the correct path.

Reply:
That's understood. The point I'm making is that your regexp shouldn't have a `?` after the port because you only want to match occurrences that have a port.